### PR TITLE
Unescaped Slashes Aint Welcome Around 'Ere

### PR DESCRIPTION
**Pull request by philsturgeon**
Seeing as unespcaped slashes are not allowed, and various tools will escape them or just not know what to do with one due to the behavior being undefined, we should probably define it.

Closes https://github.com/OAI/OpenAPI-Specification/issues/2204

Might close https://github.com/OAI/OpenAPI-Specification/issues/892

### CODE REVIEW

**Review by lornajane — APPROVED** [review]
Good clarification

**Comment by darrelmiller** [comment]
I will investigate URI Templates to identify all the allowed characters and we will update to include all disallowed chars.

File: versions/3.1.0.md [hunk 1]
```diff
@@ -75,6 +75,8 @@ Path templating refers to the usage of template expressions, delimited by curly
 
 Each template expression in the path MUST correspond to a path parameter that is included in the [Path Item](#path-item-object) itself and/or in each of the Path Item's [Operations](#operation-object).
 
+The value for these path parameters MUST NOT contain unescaped forward slashes (`/`).
```
Comment (darrelmiller) [inline versions/3.1.0.md:78]: I believe we should also outlaw ? and # as they would change the meaning of the URL.

Comment (darrelmiller) [reply versions/3.1.0.md:78]: We also need to tell people to refer back to RFC 3986 rules for URLs.

**Comment by leonmende** [comment]
Hola soy novato desde cero e cometido un sin fin de errores y al parecer e afectado a terceros sin no sé si se pueda seguir trabajando con este repocitorio estoy esperando respuesta de la plataforma ojalá y me den luz verde para seguir adelante gracias

File: versions/3.1.0.md [hunk 2]
```diff
@@ -75,6 +75,8 @@ Path templating refers to the usage of template expressions, delimited by curly
 
 Each template expression in the path MUST correspond to a path parameter that is included in the [Path Item](#path-item-object) itself and/or in each of the Path Item's [Operations](#operation-object).
 
+The value for these path parameters MUST NOT contain any unescaped "generic syntax" characters described by [RFC3986](https://tools.ietf.org/html/rfc3986#section-3): forward slashes (`/`), forward question marks (`?`), or hashes (`#`).
```
Comment (MikeRalphson) [inline versions/3.1.0.md:78]: Do we mean the "reserved characters" defined in https://tools.ietf.org/html/rfc3986#section-2.2 or something else by '"generic syntax" characters'? And "**forward**" question-marks, that's precise!

Comment (hkosova) [inline versions/3.1.0.md:78]: Consider rephrasing as an instruction for how to encode these special characters:

> The following characters MUST be percent-encoded when used in the value of a path parameter: forward slash (`/`), question mark (`?`), number sign (`#`).

(or similar)

Note that [RFC 3986](https://tools.ietf.org/html/rfc3986#section-1.2.3) refers to `#` as the "number sign", I'd suggest using the same term for consistency.

Comment (philsturgeon) [reply versions/3.1.0.md]: Reserved is more than the generic delimeters I was trying to suggest.

      reserved    = gen-delims / sub-delims

      gen-delims  = ":" / "/" / "?" / "#" / "[" / "]" / "@"

      sub-delims  = "!" / "$" / "&" / "'" / "(" / ")"
                  / "*" / "+" / "," / ";" / "="

But also maybe `[`, `]`, and `@` are ok, so I have no idea what section or word to use to describe them.

**Comment by leonmende** [comment]
Bueno de verdad que no se de donde empezar ni terminar alguien lo puedes
hacer terminar??? Y si hay ganancias los repartemo de entre los que
aportamos todos parejo cómo ven??

**Comment by MikeRalphson** [comment]
@leonmende the de facto language for this repository is English, for better or worse. Could you provide a translation with any future comments to save the majority of recipients from having to do it? I would also ask that you refrain from making general comments about contributing to the project on existing technical issues or unrelated PRs. Please raise a new issue if you have an on-topic one. "Profits" and cryptocurrencies (except where related to APIs) are not on-topic.

**Comment by philsturgeon** [comment]
Implemented all feedback:

- Removed the RFC mention
- number sign not hash
- "forward query string" lol

**Comment by leonmende** [comment]
> @leonmende the de facto language for this repository is English, for better or worse. Could you provide a translation with any future comments to save the majority of recipients from having to do it? I would also ask that you refrain from making general comments about contributing to the project on existing technical issues or unrelated PRs. Please raise a new issue if you have an on-topic one. "Profits" and cryptocurrencies (except where related to APIs) are not on-topic.

Le pido disculpas no haré más y a decir verdad creo que estoy fuera de lugar y con su permiso me retiro!!

**Comment by leonmende** [comment]
I apologize I will not do more and to tell the truth I think I am out of place and with your permission I am leaving !!

**Review by leonmende — APPROVED** [review]

**Review by darrelmiller — APPROVED** [review]

**Pull request merged**